tests/src/dir.targets

```diff
@@ -66,7 +66,6 @@
   build a run only for a higher priority testcase.
   -->
   <Target Name="Build" Condition="('$(CLRTestPriority)' &lt;= '$(CLRTestPriorityToBuild)') Or ('$(_CLRTestInServiceOfHigherPriority)')">
-    <MSBuild Projects="@(ProjectReference)" Properties="_CLRTestInServiceOfHigherPriority=true"/>
```

Comment:
You were probably getting lucky with your local build where something had built the CLRTestProjectToRun previously.  This can happen if you were building with a single node and the order of enumeration worked out or multi-nodes and you got lucky, or you didn't clean a previous build.  One thing to try would be to clean the whole output and then build just a single project that uses CLRTestProjectToRun.

The gist of this bug is that you need to avoid the pattern of calling MSBuild on another project and passing in global properties.  That is what is going to break multi-proc build and cause MSBuild to try to build the thing twice to the same output path.

I think you'll need to work with @kouvel  and @bryanar to understand what they were trying to accomplish with this change (https://github.com/wtgodbe/coreclr/commit/4cf34fe7988ad04eaa7f699a5d0faea8a8394f95) and come up with a way to do it that doesn't involve setting properties on the project reference.  As far as I can tell it looks like they were using a priority system to filter the projects and wanted to ensure a csproj with priority 2 would build if it was referenced by another one with priority 1 and they were running for pri 1.

A couple options I could imagine:
1. Always build stuff that doesn't set CLRTestProjectToRun.  Potentially setting the priority of the 
2. Instead of using _CLRTestInServiceOfHigherPriority have some property in the project like CLRTestHighestPriority that represents the highest priority of all referencing projects.